### ISSUE TITLE
clippy does not verify `#[clippy::...]` attributes

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust-clippy#14496]
### Summary

NOTE: this bug is **informational only**. Please do not make changes here until I have a chance to make an RFC for `#![register_tool]`.

In the following code, `#[clippy::here::have::some::metadata(1, 2, 3)]` is accepted by clippy without complaint. This is strange. To my knowledge, clippy does not have any attributes, only lints; but regardless it should probably warn if it doesn't recognize the attribute, because rustc will not do so for tool attributes.

I said this was informational because **people are using this** and making a change will break them. I don't want to disallow this until we have an alternative solution. Not linking because I don't want to brigade.

### Reproducer

I tried this code:

```rust
#[clippy::here::have::some::metadata(1, 2, 3)]
fn foo() {}
```

I expected to see this happen: A warning or error that the attribute is ignored or invalid.

Instead, this happened: The attribute is silently ignored.

### Version

```text
1.87.0-nightly (2025-03-28 920d95eaf23d7eb6b415)
```

**Comment by Alexendoo** [comment]
We have a few attributes and do issue warnings for unknown ones, but it looks like we only verify length 2 paths currently

https://github.com/rust-lang/rust-clippy/blob/d78622eeb82975e51d72111fd4ccb7d8a67ec2f1/clippy_utils/src/attrs.rs#L75
